### PR TITLE
Issue617/gpl 815 migration off travis

### LINKED ISSUE #617: GPL-815 Migration off travis
This represents the limber work required for https://github.com/sanger/General-Backlog-Items/issues/109

Original issue:
# Background

Travis are have changed their open-source free tier, which places heavy restrictions on our needs. We are currently on a paid plan, but are looking to migrate off.

I have made notes about the changes made to migrate Sequencescape here:
https://github.com/JamesGlover/travis_to_gh_actions

# Migrated
- [x] Sequencescape
- [x] Warren

# Still on travis, but some actions
These may already have all their actions migrated, although I think its mostly docker build files.
They should have all actions migrated, and be removed from travis
- [x] Baracoda
- [ ]  lighthouse-ui
- [ ] samples_extraction
- [ ] print_my_barcode

# Repo to migrate
- [ ] Traction
- [x] Asset Audits
- [ ] Traction-ui
- [ ] Quanthub
- [ ] Sm_workflow_lims
- [ ] sprint
- [ ] eln_pmb_bridge [Can't see on travis.com]
- [ ] mixtio
- [ ] gatekeeper
- [x] baracoda
- [ ] traction-service-automated-releases ? (What is this?)
- [ ] traction-service
- [ ] event_warehouse_ukbb
- [ ] sanger_barcode_format
- [x] event_warehouse
- [ ] unified_warehouse
- [x] limber
- [ ] pmb-client
- [x] labwhere

# Probably don't migrate
- [ ] json_api_client 
This is a fork of another gem, which brings in some fixes. We should focus of migrating to the latest version, rather than maintaining a fork.

# Repos without CI
These repos currently have no CI solution. In some cases its because one probably isn't warranted.
## CI needed?
- [ ] sequencescape-client-api
- [ ] sprint_client
- [ ] hca-print-tool
- [ ] ftpcmd
- [ ] warehouse_two

## CI probably not needed
- [ ] storelight-sql
- [ ] config
- [ ] .github
- [ ] beckman-robot-scripts
- [ ] deployment
- [ ] provision
- [ ] usg_scripts
- [ ] ss_cherrypicking (Empty repo)
- [ ] plate_barcode [ Migrate to baracoda instead ]
- [ ] General backlog items
- [ ] axlsx [Migrate to community fork instead of maintaining our own]
- [ ] label_printer [Yana's tool, mostly replaced by steve's recent stuff, not sure how you'd ci the tests]
- [ ] sequencescape_search [I don't think we use this?]
- [ ] sanger_barcode [Legacy barcode printing]
- [ ] irods_reader [Not using functionality]
- [ ] psd_logger [No longer used?]

# Repos already on GH actions
These repos either never used travis, or were migrated a while ago. I'm judging this by the presence of a .github/workflows folder
- [x] Record Loader(Never used travis)
- [x] Crawler
- [x] storelight
- [x] lighthouse
- [x] wrangler
- [x] stan-client
- [x] stan-core
- [x] automanual

### PR DESCRIPTION
**Pull request by JamesGlover**
Closes #617

Changes proposed in this pull request:

* Migrate to gh actions

### CODE REVIEW

File: .github/workflows/js_test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
```
Comment (KatyTaylor) [inline .github/workflows/js_test.yml:19]: should we be getting the node version from the .nvmrc ?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+    - name: Install
+      run: yarn install
+    - name: Run yarn test
```
Comment (KatyTaylor) [inline .github/workflows/lint.yml:35]:
```suggestion
    - name: Run yarn lint
```

File: .github/workflows/ruby_build.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: Ruby Build
+
+on:
+  release:
+    types: published
```
Comment (KatyTaylor) [inline .github/workflows/ruby_build.yml:5]: does this exclude pre-release? So would it work for UAT builds marked pre-release?

File: .github/workflows/ruby_build.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+env:
+  BUNDLE_WITHOUT: 'test lint'
```
Comment (KatyTaylor) [inline .github/workflows/ruby_build.yml:8]: what about 'develop'?

File: Gemfile [hunk 5]
```diff
@@ -2,47 +2,46 @@
...
-gem 'pmb-client', '0.1.0', github: 'sanger/pmb-client'
-gem 'sprint_client'
-
-gem 'puma'
-gem 'sanger_barcode_format', github: 'sanger/sanger_barcode_format', branch: 'development'
+  gem 'pmb-client', '0.1.0', github: 'sanger/pmb-client'
+  gem 'sprint_client'
 
+  gem 'puma'
+  gem 'sanger_barcode_format', github: 'sanger/sanger_barcode_format', branch: 'development'
```
Comment (KatyTaylor) [inline Gemfile:31]: weird that this is development not master

File: README.md [hunk 6]
```diff
@@ -1,9 +1,10 @@
 # Limber Pipeline Application
 
-[![Build Status](https://travis-ci.org/sanger/limber.svg?branch=next_release)](https://travis-ci.org/sanger/limber)
+![Linting](https://github.com/JamesGlover/limber/workflows/Linting/badge.svg)
+![Ruby RSpec Tests](https://github.com/JamesGlover/limber/workflows/Ruby%20RSpec%20Tests/badge.svg)
+![Javascript testing](https://github.com/JamesGlover/limber/workflows/Javascript%20testing/badge.svg)
```
Comment (KatyTaylor) [inline README.md:5]: similar to on other PR - referring to your repo - should probably refer to Sanger

File: Gemfile (hunk 5 above)
Comment (JamesGlover) [reply Gemfile:31]: Yeah, it was probably a case of me wanted to run with some improvements, but they hadn't yet been merged to master. We should probably just add it to ruby-gems instead,

File: README.md (hunk 6 above)
Comment (JamesGlover) [reply README.md:5]: Ooh, good catch.

File: .github/workflows/js_test.yml (hunk 1 above)
Comment (JamesGlover) [reply .github/workflows/js_test.yml:19]: Oh, hadn't realised that was there. Yes, we probably should.

File: .github/workflows/ruby_build.yml (hunk 3 above)
Comment (JamesGlover) [reply .github/workflows/ruby_build.yml:5]: It includes pre-releases, just not draft releases.

File: .github/workflows/ruby_build.yml (hunk 4 above)
Comment (JamesGlover) [reply .github/workflows/ruby_build.yml:8]: development includes the gems that are part of the asset pipeline, so are needed to build the javascript

File: .github/workflows/ruby_build.yml (hunk 3 above)
Comment (KatyTaylor) [reply .github/workflows/ruby_build.yml:5]: Ah OK no probs, I hadn't really noticed drafts were a thing

File: .github/workflows/ruby_build.yml (hunk 4 above)
Comment (KatyTaylor) [reply .github/workflows/ruby_build.yml:8]: OK thanks. Why are they in development rather than default then?

File: .github/workflows/lint.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true # Runs bundle install and caches gems. See the ruby_test.yml
```
Comment (KatyTaylor) [inline .github/workflows/lint.yml:19]: I made a comment about this in the other pull request - that `ruby_test.yml` is meaningless in these repos - so feel free to change in all the places where you've copy-pasted, or not!

**Review by KatyTaylor — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 7 above)
Comment (JamesGlover) [reply .github/workflows/lint.yml]: Oh bah... removed.

**Pull request merged**